components/configuration-mgt/org.wso2.carbon.identity.configuration.mgt.core/src/main/java/org/wso2/carbon/identity/configuration/mgt/core/dao/impl/ConfigurationDAOImpl.java

```diff
@@ -621,8 +664,17 @@ public void addResource(Resource resource) throws ConfigurationManagementExcepti
                             not create files. It is allowed to create a resource without files or attributes in order
                             to allow file upload after resource creation.
                             */
-                            preparedStatement.setBoolean(++initialParameterIndex, false);
-                            preparedStatement.setBoolean(++initialParameterIndex, isAttributeExists);
+                            try {
+                                if (isOracleDB() || isMSSqlDB()) {
+                                    preparedStatement.setInt(++initialParameterIndex, 0);
+                                    preparedStatement.setInt(++initialParameterIndex, isAttributeExists ? 1 : 0);
+                                } else {
+                                    preparedStatement.setBoolean(++initialParameterIndex, false);
+                                    preparedStatement.setBoolean(++initialParameterIndex, isAttributeExists);
+                                }
+                            } catch (DataAccessException e) {
+                                throw new RuntimeException(e);
+                            }
```

Comment:
@emswbandara please check

Reply:
why we throw runtime exception here? can't we throw a server exception using handleServerException ?